operator/controllers/lifecycle/keptnappversion/controller_test.go

```diff
@@ -185,7 +186,13 @@ func setupReconciler() (*KeptnAppVersionReconciler, chan string, *fake.ITracerMo
 		UnbindSpanFunc: func(reconcileObject client.Object, phase string) error { return nil },
 	}
 
-	fakeClient := fake.NewClient()
+	workloadInstanceIndexer := func(obj client.Object) []string {
+		workloadInstance, _ := obj.(*lfcv1alpha3.KeptnWorkloadInstance)
+		return []string{workloadInstance.Spec.AppName}
+	}
+
+	fake.SetupSchemes()
+	fakeClient := k8sfake.NewClientBuilder().WithObjects(objs...).WithScheme(scheme.Scheme).WithObjects().WithIndex(&lfcv1alpha3.KeptnWorkloadInstance{}, "spec.app", workloadInstanceIndexer).Build()
```

Comment:
what if we modify our common package fake client to also be able to add indexes? this way we do not need to add scheme and build client every time and close would look a bit simpler

Reply:
I personally prefer having the possibility to define what I need and not to use the fakeClient. If we would add the index there, we needed additionally 3 params and in most cases it's not needed at all.

Reply:
mm I ment a second function such as getWithIndex, but indeed we do not need it now so it's ok :)